tests/main.cpp

```diff
@@ -6,3 +6,15 @@
 
 #define CATCH_CONFIG_MAIN
 #include "catch.hpp"
+
+#if WITEST_ADDRESS_SANITIZER
+extern "C" __declspec(dllexport)
+const char* __asan_default_options()
+{
+    return
+        // Tests validate OOM, so this is expected
+        "allocator_may_return_null=1"
+        // Some structs in Windows have dynamic size where we over-allocate for extra data past the end

```

Comment:
A-San complains that we are calling `delete` such that the memory allocated is > `sizeof(T)`